From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Thu, 24 Nov 2022 21:26:57 +0100
Subject: [PATCH] Fiddle project setup

Sets up the `fiddle-server` module build file correctly:
* Remove Paper inter-module dependencies
* Create a constant for the `fiddle-api` module name and depend on it
* Re-add dependency on Paper MojangAPI (but externally this time)

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/build.gradle.kts b/build.gradle.kts
index 58da26ad2f128ba0b66f86820f60853f4be352f0..295855fca149e12c6bf976c6ff577084722c62c4 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -12,9 +12,17 @@ configurations.named(log4jPlugins.compileClasspathConfigurationName) {
 }
 val alsoShade: Configuration by configurations.creating
 
+val apiProjectName = "fiddle-api" // Fiddle - project setup
+
 dependencies {
-    implementation(project(":paper-api"))
-    implementation(project(":paper-mojangapi"))
+    // Fiddle start - project setup
+    // Depend on own API
+    implementation(project(":$apiProjectName"))
+    // Depend on Paper MojangAPI
+    implementation("io.papermc.paper:paper-mojangapi:1.20.3-R0.1-SNAPSHOT") {
+        exclude("io.papermc.paper", "paper-api")
+    }
+    // Fiddle end - project setup
     // Paper start
     implementation("org.jline:jline-terminal-jansi:3.21.0")
     implementation("net.minecrell:terminalconsoleappender:1.3.0")
